packages/ai/src/chains/operations/index.ts

```diff
@@ -0,0 +1,151 @@
+import { z } from "zod";
+import type { Model as BaseModel, ModelMessage, Chain } from "../../types";
+import { formatPrompt } from "../../utils/format-prompt";
+import { prompt as promptSystemTemplate } from "./__generated__/edit.system.prompt";
+import { prompt as promptUserTemplate } from "./__generated__/edit.user.prompt";
+
+import {
+  AiOperationsSchema,
+  aiToWs,
+  WsOperationsSchema,
+  type WsOperations,
+} from "./operations";
+import { zodToJsonSchema } from "zod-to-json-schema";
+import { postprocessTemplate } from "../../utils/jsx-to-template";
+
+export * from "./operations";
+
+/**
+ * Operations Chain.
+ *
+ * Given a description, available components and an existing instance as JSX and CSS,
+ * it generates a series of edit operations to fulfill an edit request coming from the user.
+ */
+
+export const ContextSchema = z.object({
+  prompt: z.string().max(1200).describe("Edit request from the user"),
+  components: z.array(z.string()).describe("Available Webstudio components"),
+  jsx: z.string().describe("Input JSX to edit"),
+});
+export type Context = z.infer<typeof ContextSchema>;
+
+export const ResponseSchema = WsOperationsSchema;
+export type Response = z.infer<typeof ResponseSchema>;
+
+export const createChain = <ModelMessageFormat>(): Chain<
+  BaseModel<ModelMessageFormat>,
+  Context,
+  Response
+> =>
+  async function chain({ model, context }) {
+    const { prompt, components, jsx } = context;
+
+    const llmMessages: ModelMessage[] = [];
+    const tokens = { prompt: 0, completion: 0 };
+
+    const operationsSchema = zodToJsonSchema(
+      AiOperationsSchema.element,
+      "AiOperationsSchema"
+    );
+
+    const systemMessage: ModelMessage = [
+      "system",
+      formatPrompt(
+        {
+          operationsSchema: JSON.stringify(operationsSchema),
+          components: components
+            .map((name) =>
+              name.replace(
+                "@webstudio-is/sdk-components-react-radix:",
```

Comment:
Uhh this belongs somwere else, radix is just a package, like any user-installable one, it shouldn't know directly about it